File: android/app/src/main/kotlin/xyz/alaniz/aaron/lightsaber/ui/lightsaber/LightsaberModels.kt
package xyz.alaniz.aaron.lightsaber.ui.lightsaber

import androidx.compose.ui.graphics.Color

data class LightsaberState(
    val bladeColor: Color,
    val bladeState: BladeState,
    val onEvent: (LightsaberEvent) -> Unit
)

sealed interface LightsaberEvent {
    object SettingsSelected : LightsaberEvent
    object LightsaberActivating : LightsaberEvent
    object LightsaberDeactivating : LightsaberEvent
    object LightsaberActivated : LightsaberEvent
    object LightsaberDeactivated : LightsaberEvent
}

enum class BladeState {
    Initializing,
    Activating,
    Activated,
    Deactivating,
    Deactivated
}

File: ios/app/Sources/LightsaberApp/LightsaberModels.swift
import SwiftUI

struct LightsaberState {
    let bladeColor: Color
    let bladeState: BladeState
    let onEvent: (LightsaberEvent) -> Void
}

enum LightsaberEvent {
    case settingsSelected
    case lightsaberActivating
    case lightsaberDeactivating
    case lightsaberActivated
    case lightsaberDeactivated
}

enum BladeState {
    case initializing
    case activating
    case activated
    case deactivating
    case deactivated
}
